protected/humhub/modules/tour/controllers/TourController.php

```diff
@@ -41,55 +37,21 @@ protected function getAccessRules()
      */
     public function actionTourCompleted()
     {
-        /** @var Module $module */
-        $module = Yii::$app->getModule('tour');
+        // get Tour ID from the completed tour
+        $tourId = Yii::$app->request->post('tour_id');
 
-        // get section parameter from completed tour
-        $section = Yii::$app->request->post('section');
-
-        if (!in_array($section, $module->acceptableNames)) {
+        if (!TourConfig::IsCurrentRouteAcceptable($tourId)) {
             return;
         }
 
         // set tour status to seen for current user
-        Yii::$app->getModule('tour')->settings->user()->set($section, 1);
+        Yii::$app->getModule('tour')->settings->user()->set($tourId, true);
```

Comment:
Just to be sure, an invalid `tourId` POST param will never be stored as settings because isCurrentRouteAcceptable?